snapcraft/tests/test_commands_push.py

```diff
@@ -245,3 +247,80 @@ def test_push_and_release_a_snap_to_N_channels(self):
         mock_upload.assert_called_once_with('my-snap-name', snap_file)
         mock_release.assert_called_once_with('my-snap-name', 9,
                                              ['edge', 'beta', 'candidate'])
+
+    def test_push_revision_cached_experimental_deltas_enabled(self):
```

Comment:
@sergiusens PR updated using test scenarios (thanks for that, that's much nicer!).